packages/autocomplete-plugin-tags/src/createTags.ts

```diff
@@ -0,0 +1,72 @@
+import { BaseItem } from '@algolia/autocomplete-core';
+import { createRef } from '@algolia/autocomplete-shared';
+
+import { CreateTagsPluginParams } from './createTagsPlugin';
+import { BaseTag, Tag } from './types';
+
+export type OnTagsChangeParams<TTag> = {
+  prevTags: Array<Tag<TTag>>;
+  tags: Array<Tag<TTag>>;
+};
+
+type OnTagsChangeListener<TTag> = (params: OnTagsChangeParams<TTag>) => void;
+
+type CreateTagsParams<TTag, TItem extends BaseItem> = Pick<
+  CreateTagsPluginParams<TTag, TItem>,
+  'initialTags'
+>;
+
+export function createTags<TTag, TItem extends BaseItem>({
+  initialTags = [],
+}: CreateTagsParams<TTag, TItem>) {
+  const tagsRef = createRef(toTags(initialTags));
+  const onChangeListeners: Array<OnTagsChangeListener<TTag>> = [];
+
+  function toTags<TTag>(baseTags: Array<BaseTag<TTag>>): Array<Tag<TTag>> {
+    return baseTags.map((baseTag) => {
+      const tag = {
+        ...baseTag,
+        remove() {
+          const prevTags = tagsRef.current.slice();
+          const newTags = tagsRef.current.filter(
+            (tagRef) => tag !== ((tagRef as unknown) as Tag<TTag>)
+          );
+
+          tagsRef.current = newTags;
+          onChangeListeners.forEach((listener) =>
+            listener({ prevTags, tags: newTags })
+          );
+        },
+      };
+
+      return tag;
+    });
+  }
+
+  return {
+    get() {
+      return tagsRef.current;
+    },
+    set(baseTags: Array<BaseTag<TTag>>) {
+      const prevTags = tagsRef.current.slice();
+      const newTags = toTags(baseTags);
+
+      tagsRef.current = newTags;
+      onChangeListeners.forEach((listener) =>
+        listener({ prevTags, tags: tagsRef.current })
```

Comment:
Line 37 (https://github.com/algolia/autocomplete/pull/645/files#diff-896040d94c9485280e63a10fe79eb3a33938c46a11d42289e62c8d117917c2b8R37) we pass `newTags` to the `tags` argument, here we pass `tagsRef.current`. Can we provide the same value in both line 37 and 56 for consistency?

Reply:
Yup, makes sense. I dropped `newTags` since it was never complex logic anyway, it avoid having two variables referring to the same value and confusion over which one to use. Addressed in 3e3dfb48.